compiler/rename/RnExpr.hs

```diff
@@ -2198,7 +2198,7 @@ getMonadFailOp
 -- rec. Hence, compared to the ghc implementation, the arguments are switched here.
 mkSetField :: Name -> LHsExpr GhcRn -> Located FieldLabelString -> LHsExpr GhcRn -> HsExpr GhcRn
 mkSetField set_field a (L _ field) b =
-  genHsApp (genHsApp (genHsVar set_field `genAppType` genHsTyLit field)  b) a
+  genHsPar (genHsApp (genHsApp (genHsVar set_field `genAppType` genHsTyLit field) b) a)
```

Comment:
I think for the example I saw, we should also add a `genHsPar` around `b` here:

```suggestion
  genHsPar (genHsApp (genHsApp (genHsVar set_field `genAppType` genHsTyLit field) (genHsPar b)) a)
```

Possibly around `a` as well? I'm not sure.

Reply:
that's a good catch, thank you! 

AFAICT we don't need one for `a`, since otherwise `with` syntax forces the user to parenthesize the thing being with-d, e.g.

```haskell
foo : (Text -> (Int, Bool)) -> (Int, Bool)
foo mkTup = (mkTup "bar") with
  _2 = not (mkTup "baz")._2
```

if `(mkTup "bar")` weren't parenthesized, the `with`-expression wouldn't parse.

Reply:
That makes sense! :)